src/components/search-supplier-modal/SearchSupplierModal.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, useEffect } from 'react';
+import { useSuppliers } from '../../hooks/useSuppliers';
+import { initializeDatabase } from '../../db/indexedDb';
+import './SearchSupplierModal.css';
+import { Supplier } from '@/types/types';
+import Table from '../../components/table/Table';
+
+type SupplierSearchModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onSelect: (supplier: Supplier) => void;
+};
+
+const SupplierSearchModal: React.FC<SupplierSearchModalProps> = ({
+  isOpen,
+  onClose,
+  onSelect,
+}) => {
+  const { searchSuppliers } = useSuppliers();
+  const [searchResults, setSearchResults] = useState<Supplier[]>([]);
+  const [searchCriteria, setSearchCriteria] = useState({
+    name: '',
+    index: '',
+    city: '',
+  });
+  const [initialResults, setInitialResults] = useState<Supplier[]>([]);
+  const [selectedSupplier, setSelectedSupplier] = useState<Supplier | null>(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      const initializeAndFetch = async () => {
+        await initializeDatabase();
+        const results = await searchSuppliers({});
```

Comment:
please do not fetch initial suppliers. In this case it is not an issue since there are only three of them, but think about if there are 100.000 suppliers in database, and you fetch all of them in once on frontend, it is not good for performance. Initially you do not need to fetch anything but only on search.